core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -327,6 +334,58 @@ public synchronized void losePrimacy() {
     LOG.info("Raft server successfully restarted");
   }
 
+  @Override
+  public synchronized void checkpoint() throws IOException {
+    mSnapshotAllowed.set(true);
+    CopycatClient client = createAndConnectClient();
```

Comment:
this should be inside the `try` - otherwise if we throw an exception, we won't set `mSnapshotAllowed` back to `false`

Reply:
Move inside the try finally. Thanks